### ISSUE TITLE
the "Grouping to matrix" transformation duplicates values

### ISSUE THREAD

**Issue by IlyaDeskov** [grafana/grafana#65248]
**What happened**:
If I want to use "grouping to matrix" transformation for data like (time column is hidden by other transforms, only aggregated data here):
<table>
<tr><td>A</td><td>B</td><td>C</td></tr>
<tr><td>foo</td><td>1</td><td>what</td></tr>
<tr><td>foo</td><td>2</td><td>some</td></tr>
<tr><td>bar</td><td>3</td><td>what</td></tr>
<tr><td>bar</td><td>4</td><td>some</td></tr>
<tr><td>buz</td><td>5</td><td>what</td></tr>
<tr><td>buz</td><td>6</td><td>some</td></tr>
</table>

I select "A" as Column, "C" as Row and "B" as Cell Value

and I got this:

<table>
<tr><td>C\A</td><td>foo</td><td>bar</td><td>baz</td></tr>
<tr><td>what</td><td>1</td><td>1</td><td>1</td></tr>
<tr><td>some</td><td>2</td><td>2</td><td>2</td></tr>
<tr><td>what</td><td>3</td><td>3</td><td>3</td></tr>
<tr><td>some</td><td>4</td><td>4</td><td>4</td></tr>
<tr><td>what</td><td>5</td><td>5</td><td>5</td></tr>
<tr><td>some</td><td>6</td><td>6</td><td>6</td></tr>
</table>

**What you expected to happen**:
but I expect to obtain this:
<table>
<tr><td>C\A</td><td>foo</td><td>bar</td><td>baz</td></tr>
<tr><td>what</td><td>1</td><td>3</td><td>5</td></tr>
<tr><td>some</td><td>2</td><td>4</td><td>6</td></tr>
</table>

**How to reproduce it (as minimally and precisely as possible)**:

1. Use grafana Table panel and influxdb
2. Select data with grouping by 2 tags (grouping should bring unique rows. Do not group by time)
3. Use "organize fields" to hide the time column
4. Use "Grouping to matrix" to try to pivot data with unique cols and rows

**Anything else we need to know?**:

**Environment**:
- Grafana version:9.0.7
- Data source type & version: InfluxDB 
- OS Grafana is installed on: Debian 10
- User OS & Browser: Win10 & Firefox Developer Ed 112.0b3 (64-разрядный)
- Grafana plugins:
- Others:

**Comment by zuchka** [comment]
could you please copy and paste all relevant data using the `panel --> more --> get help` workflow in the panel's dropdown menu? This will help up mock up your data and reproduce your issue. It is important to our teams that issues get verified as reproducible before we formally mark them as bugs. Thank you!

here are more details and steps for copying your data over to this issue:

https://grafana.com/docs/grafana/latest/troubleshooting/send-panel-to-grafana-support/

![CleanShot 2023-01-31 at 14 40 21](https://user-images.githubusercontent.com/37156449/215916832-8d29d39c-57a1-40a7-a6e0-bfe584ea8bd4.gif)

**Comment by IlyaDeskov** [comment]
@zuchka Hello, Matt
My version (9.0) hasn't this feature, and selection result contains sensitive data. I can add panel JSON to this task. Would it be helpful?

**Comment by IlyaDeskov** [comment]
Try to attach the file with panel JSON via mail reply. 24.03.2023, 06:33, "matt abrams" ***@***.***>: could you please copy and paste all relevant data using the panel --> more --> get help workflow in the panel's dropdown menu? This will help up mock up your data and reproduce your issue. It is important to our teams that issues get verified as reproducible before we formally mark them as bugs. Thank you!here are more details and steps for copying your data over to this issue:https://grafana.com/docs/grafana/latest/troubleshooting/send-panel-to-grafana-support/—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>

**Comment by zuchka** [comment]
not sure it came through @IlyaDeskov

**Comment by IlyaDeskov** [comment]
Yes, I tried to reply via email and attached the file "panel with matrix duplication.json"
